CMakeLists.txt

```diff
@@ -0,0 +1,984 @@
+cmake_minimum_required(VERSION 3.25)
```

Comment:
While I would love to move the project fully to cmake, best we dont couple cmake files into this pr if possible.
(If you are willing to help port to cmake; I would hands down love it though)
Could you drop this file from this pr though :)

Reply:
Hahahaha, oh god i have no idea about CMake tbh, i just had to add this so my IDE (CLion) would be able to understand the project :D Will drop it.